pyiron/atomistics/structure/atoms.py

```diff
@@ -1118,24 +1118,75 @@ def get_extended_positions(self, width):
         v_repeated = v_repeated[check_dist]
         return v_repeated, indices
 
+    def get_number_of_neighbors_in_sphere(
+            self,
+            cutoff_radius=10,
+            num_neighbors=None,
+            tolerance=2,
+            id_list=None,
+            boundary_width_factor=1.2,
+    ):
+        """
+        Function to compute the number of neighbors in a sphere around each atom.
+        Args:
+            cutoff_radius (float): Upper bound of the distance to which the search must be done
+            num_neighbors (int): maximum number of neighbors found
+            tolerance (int): tolerance (round decimal points) used for computing neighbor shells
+            id_list (list): list of atoms the neighbors are to be looked for
+            boundary_width_factor (float): width of the layer to be added to account for pbc.
+
+        Returns:
+            maximum number of neigbors found in the sphere of radius cutoff_radius (<= num_neighbors if specified)
+        """
+        if num_neighbors is not None:
+            neigh = self._get_neighbors(
+                num_neighbors=num_neighbors,
+                t_vec=False,
+                tolerance=tolerance,
+                id_list=id_list,
+                cutoff_radius=cutoff_radius,
+                boundary_width_factor=boundary_width_factor,
+            )
+            num_neighbors = max(len(elem) for elem in neigh.indices)
+        else:
+            num_neighbors = int(0.5*cutoff_radius**3)
```

Comment:
I guess it makes more sense to make it dependent on the density, rather than just the length. How about something like `num_neighbors = int(1.1*cutoff_radius**3/self.get_volume(per_atom=True))`?

Reply:
Sorry since you provide `id_list` (which is a good idea), the volume per atom should rather be:

```
volume_per_atom = self.get_volume(per_atom=True)
if id_list is not None:
    volume_per_atom = self.get_volume()/len(id_list)
```

Reply:
To make it dependent on the density is something nice, however, the volume within the sphere would be 
`4/3 * np.pi * cutuff_radius**3` and thus 
```
volume_per_atom = self.get_volume(per_atom=True)
if id_list is not None:
    volume_per_atom = self.get_volume()/len(id_list)
num_neighbors = int(1.1*4./3.*np.pi*cutoff_radius**3/volume_per_atom)
```